pkg/provisioners/message_receiver.go

```diff
@@ -99,17 +88,38 @@ func (r *MessageReceiver) stop(srv *http.Server) {
 	}
 }
 
+// handler creates the http.Handler used by the http.Server started in MessageReceiver.Run.
+func (r *MessageReceiver) handler() http.Handler {
+	return http.HandlerFunc(func(res http.ResponseWriter, req *http.Request) {
+		if req.URL.Path != "/" {
+			res.WriteHeader(http.StatusNotFound)
+			return
+		}
+		if req.Method != http.MethodPost {
+			res.WriteHeader(http.StatusMethodNotAllowed)
+			return
+		}
+
+		r.HandleRequest(res, req)
+	})
+}
+
 // HandleRequest is an http Handler function. The request is converted to a
 // Message and emitted to the receiver func.
 //
 // The response status codes:
-//   202 - the message was sent to subscibers
+//   202 - the message was sent to subscribers
 //   404 - the request was for an unknown channel
 //   500 - an error occurred processing the request
 func (r *MessageReceiver) HandleRequest(res http.ResponseWriter, req *http.Request) {
 	host := req.Host
 	r.logger.Infof("Received request for %s", host)
-	channel := ParseChannel(host)
+	channel, err := ParseChannel(host)
+	if err != nil {
+		r.logger.Info("Could not extract channel", zap.Error(err))
+		res.WriteHeader(http.StatusInternalServerError)
```

Comment:
It might be useful to add some indication to the Response so that it can be seen what the error actually was (not just the code, but some indication that the system seems to be in a bad state, but there's no indication necessarily what the problem is).